### PR TITLE
refactor(*): upgrade all dependencies

### PR DESCRIPTION
**Pull request by willsoto**
Also upgrade infrastructure

@hakimio Tests are not running at all for me locally (`ng test ng-chartist --watch=false --browsers=ChromeHeadless` just exits without doing anything). Do you know of any breaking changes there between angular v8 and v12 😅

### CODE REVIEW

**Comment by hakimio** [comment]
@willsoto `async()` was renamed to `waitForAsync()`. Not sure if there is anything else.
[source](https://angular.io/api/core/testing/async)

File: package.json [hunk 1]
```diff
@@ -1,110 +1,120 @@
...
+    "bracketSpacing": true,
+    "semi": true,
+    "singleQuote": true
+  },
+  "devDependencies": {
+    "@angular-devkit/build-angular": "~12.0.2",
+    "@angular-devkit/build-ng-packagr": "~0.1002.0",
+    "@angular/animations": "~12.0.2",
+    "@angular/cli": "~12.0.2",
+    "@angular/common": "~8.1.0",
```
Comment (hakimio) [inline package.json:65]: Why is this still v8?

Comment (willsoto) [reply package.json:65]: Not sure how this didn't get picked up...I'll upgrade

File: package.json [hunk 2]
```diff
@@ -1,110 +1,120 @@
...
+    "@angular/compiler": "~12.0.2",
+    "@angular/compiler-cli": "~12.0.2",
+    "@angular/core": "~8.1.0",
```
Comment (hakimio) [inline package.json:68]: And this? Did you go through all dependencies?

Comment (willsoto) [reply package.json:68]: I didn't do it manually, I used a [tool](https://github.com/raineorshine/npm-check-updates). I wonder why these weren't picked up

**Comment by hakimio** [comment]
- Other dependencies not updated:
  - @types/chartist
  - chartist
  - tslint 
- might be a good idea to [migrate to eslint](https://github.com/angular-eslint/angular-eslint)
- `@angular-devkit/build-ng-packagr` has been deprecated and should be removed
- rxjs v7 is not compatible with Angular v12. It should be left at v6.

**Comment by hakimio** [comment]
After correcting package versions and replacing library builder from `@angular-devkit/build-ng-packagr:build` to `@angular-devkit/build-angular:ng-packagr`, everything seems to work fine. It builds and tests succeed.

**Comment by willsoto** [comment]
@hakimio are you able to push up those changes to this branch? It will be quicker for me that way to cut a release

**Comment by hakimio** [comment]
Forgot to commit `package-lock.json`. Feel free to do that.

**Comment by hakimio** [comment]
Sure.

**Comment by willsoto** [comment]
Having some issues publishing that I need to look into but I'll try and cut a release today

**Pull request merged**